parquet/src/main/java/org/apache/iceberg/parquet/ApplyNameMapping.java

```diff
@@ -39,9 +39,15 @@ class ApplyNameMapping extends ParquetTypeVisitor<Type> {
   private static final String MAP_VALUE_NAME = "value";
   private final NameMapping nameMapping;
   private final Deque<String> fieldNames = Lists.newLinkedList();
+  private final boolean caseInsensitiveLookup;
 
   ApplyNameMapping(NameMapping nameMapping) {
+    this(nameMapping, false);
```

Comment:
I think it would be cleaner to create a class `ApplyNameMappingCaseInsensitive` that extends `ApplyNameMapping` with the only change in `protected String[] currentPath()`. That way we will never have conflict in this class.